source/framework/vx_producer.c

```diff
@@ -0,0 +1,1336 @@
+                &num_items,
+                &ipc_message_parent,
+                (tivx_utils_ref_ipc_msg_t*)&ipc_message_item[0]);
+
+            buffer_desc_msg->num_items = num_items;
+
+            // send object array items data, if present
+            for (uint32_t j = 0; j < num_items; j++)
+            {
+                buffer_desc_msg->item_index = j;
+                memcpy(&buffer_desc_msg->ipc_msg, (void*)&ipc_message_item[j], sizeof(tivx_utils_ref_ipc_msg_t));
+
+                VX_PRINT(
+                    VX_ZONE_INFO,
+                    "PRODUCER %s: [VX_MSGTYPE_REF_BUF] sending objarray element %d with fd count %d\n",
+                    producer->name,
+                    j,
+                    buffer_desc_msg->ipc_msg.numFd);
+#ifdef defined(QNX)
+                status = socket_write(client->socket_fd, (uint8_t*)buffer_desc_msg, NULL, 0);
+#elif defined(x86_64)
+                status = socket_write(
+                    client->socket_fd,
+                    (uint8_t*)buffer_desc_msg,
+                    (int32_t*)buffer_desc_msg->ipc_msg.fd,
+                    buffer_desc_msg->ipc_msg.numFd);
+#endif
+                if (status != SOCKET_STATUS_OK)
+                {
+                    VX_PRINT(
+                        VX_ZONE_ERROR,
+                        "PRODUCER %s: send_reference_info() failed while sending socket message\n",
+                        producer->name);
+                    return status;
+                }
+            }
+        }
+        else
+        {
+            framework_status = tivx_utils_export_ref_for_ipc_xfer(producer->refs[i].ovx_ref, &ipc_message_parent);
+        }
+
+        if (framework_status != VX_SUCCESS)
+        {
+            VX_PRINT(VX_ZONE_ERROR, "PRODUCER: export_ref_for_ipc_xfer() failed for buffer [%d]\n", i);
+            break;
+        }
+        else
+        {
+            VX_PRINT(
+                VX_ZONE_INFO,
+                "PRODUCER %s: export of buffer successfull: %d of total: %d\n",
+                producer->name,
+                i + 1,
+                producer->numBufferRefsExport);
+        }
+
+        VX_PRINT(
+            VX_ZONE_INFO,
+            "PRODUCER %s: Sending [VX_MSGTYPE_REF_BUF] for buffer %d of type %d\n",
+            producer->name,
+            i,
+            ref_type);
+
+        if (i == (producer->numBufferRefsExport - 1))
+        {
+            VX_PRINT(
+                VX_ZONE_INFO, "PRODUCER %s: number of objects to exchange reached, set last object\n", producer->name);
+            buffer_desc_msg->last_reference = 1;
+        }
+
+        // send reference data, for object array this is final metadata
+        buffer_desc_msg->item_index = 0; // used only for object array items
+        memcpy(&buffer_desc_msg->ipc_msg, &ipc_message_parent, sizeof(tivx_utils_ref_ipc_msg_t));
+#ifdef defined(QNX)
+        status = socket_write(client->socket_fd, (uint8_t*)buffer_desc_msg, NULL, 0);
+#elif defined(x86_64)
```

Comment:
why not LINUX here? we have to choose.
Socket are for Linux, if this is linux arm or x86, it does not make a difference
QNX works with IPPC

Reply:
Done